packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -319,17 +341,66 @@ class AndroidCameraCameraX extends CameraPlatform {
         .bindToLifecycle(cameraSelector!, <UseCase>[preview!]);
   }
 
-  /// Unbinds [preview] instance to camera lifecycle controlled by the
-  /// [processCameraProvider].
-  Future<void> _unbindPreviewFromLifecycle() async {
-    final bool previewIsBound = await processCameraProvider!.isBound(preview!);
-    if (preview == null || !previewIsBound) {
+  /// Configures and binds [imageAnalysis] instance to camera lifecycle
+  /// controlled by the [processCameraProvider].
+  Future<void> _configureAndBindImageAnalysisToLifecycle() async {
+    assert(processCameraProvider != null);
+    assert(cameraSelector != null);
+
+    if (imageAnalysis != null &&
+        await processCameraProvider!.isBound(imageAnalysis!)) {
+      // imageAnalysis already configured and bound to lifecycle.
       return;
     }
 
+    // TODO(camsim99): Support resolution configuration.
+    // Defaults to YUV_420_888 image format.
+    imageAnalysis = createImageAnalysis(null);
+    imageAnalysis!.setAnalyzer();
+
+    // TODO(camsim99): Reset live camera state observers here when
+    // https://github.com/flutter/packages/pull/3419 lands.
+    camera = await processCameraProvider!
+        .bindToLifecycle(cameraSelector!, <UseCase>[imageAnalysis!]);
+  }
+
+  /// Unbinds [useCase] from camera lifecycle controlled by the
+  /// [processCameraProvider].
+  Future<void> _unbindUseCaseFromLifecycle(UseCase useCase) async {
```

Comment:
Refactored this, but didn't end up using. Left it because I still think it works for `UseCase`s in general.

Reply:
I don't think we have a repo wide guidance since we can't enforce it with a lint, but my preference is also to remove it if is not currently used. Especially if it is private. One exception is for placeholders in public interfaces: https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart#L43